### PR TITLE
Make api end point configurable

### PR DESCRIPTION
**Pull request by shariq1501**
Configure api end-point.

### CODE REVIEW

File: src/frontend/src/app/settings/settings.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <label class='label'>Type</label>
+      <div class='select'>
+        <select id="protocol">
+            <option>http</option>
+            <option>https</option>
+          </select>
+      </div>
+    </div>
+    <div class='column'>
+      <label class='label'>IP Address</label>
```
Comment (praveen-garg) [inline src/frontend/src/app/settings/settings.component.html]: Host IP or Address

File: src/frontend/src/app/settings/settings.component.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <input id='ip' class='input' type='text' placeholder='0.0.0.0'>
+    </div>
+     <div class='column'>
+      <label class="label">Port</label>
+      <input id='port' class='input' type='numbe' maxlength="4" placeholder='8082'>
+    </div>
+  </div>
+  <div class="columns">
+     <div class='column'>
+        <button class='button is-primary is-pulled-right' (click)='resetEndPoint()'>Reset URL</button>
```
Comment (praveen-garg) [inline src/frontend/src/app/settings/settings.component.html]: Set the URL and Restart

File: src/frontend/src/app/settings/settings.component.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+})
+export class SettingsComponent implements OnInit {
+  @Output() toggle: EventEmitter<any> = new EventEmitter();
+  constructor(private router: Router) { }
+
+  ngOnInit() { }
+
+  public resetEndPoint() {
+    const protocolField = <HTMLSelectElement>document.getElementById('protocol');
+    const ipField = <HTMLInputElement>document.getElementById('ip');
```
Comment (praveen-garg) [inline src/frontend/src/app/settings/settings.component.ts]: hostField host can have IP or other valid address

File: src/frontend/src/app/settings/settings.component.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    const portField = <HTMLInputElement>document.getElementById('port');
+    const ip_address = protocolField.value + '://' + ipField.value + ':' + portField.value + '/foglamp/';
```
Comment (praveen-garg) [inline src/frontend/src/app/settings/settings.component.ts]: endpoint =

File: src/frontend/src/app/settings/settings.component.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    localStorage.setItem('API_END_POINT', ip_address);
+    console.log('End Point', localStorage.getItem('API_END_POINT'));
+    this.router.navigate(['']);
+    window.location.reload();
```
Comment (praveen-garg) [inline src/frontend/src/app/settings/settings.component.ts]: is this angular way?

**Review by praveen-garg — CHANGES_REQUESTED** [review]
LGTM from dry run review; need minor chnages and testing as fresh and rerun with manual deletion of localStorage on ff, safari and chrome

**Comment by MonikaSharma06** [comment]
Tested on Safari, Chrome and Firefox. All things are working fine except one thing, in Safari and Firefox, after clicking the button `Set the URL & Restart` it is not redirecting to Home Page (Dashboard). It remains on Setting Tab.

**Comment by shariq1501** [comment]
Route redirection issue fixed on Firefox and Safari browser.

**Comment by shariq1501** [comment]
@praveen-garg You can merge this PR now.

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**